### ISSUE TITLE
Crash on Linux Server

### ISSUE THREAD

**Issue by AtlanticCityRoleplay** [ziggi/FCNPC#295]
```
=========================== FCNPC Exception Handler ============================
Signal 11 (Segmentation fault), address is 5dc30815 from 80b0036

Backtrace:
  #0 in plugins/FCNPC.so(_ZN17CExceptionHandler24ExceptionHandlerCallbackEiP9siginfo_tPv+0x126) [0xeee5f6e6]
  #1 in [0xf7f56b70]
  #2 in ./samp03svr() [0x80b0036]

Information:
  System: GNU/Linux
  SA-MP: 0.3.7 R2
  FCNPC: 2.0.5

Functions:
  FUNC_CPlayerPool__DeletePlayer: 0x80d0a90
  FUNC_CPlayer__Kill: 0x80cb220
  FUNC_CPlayer__EnterVehicle: 0x80cc1c0
  FUNC_CPlayer__ExitVehicle: 0x80cc340
  FUNC_CPlayer__SpawnForWorld: 0x80ccfc0
  FUNC_GetVehicleModelInfo: 0x80d5e90
  FUNC_CConsole__GetIntVariable: 0x80a0070
  FUNC_ClientJoin_RPC: 0x80b0030

Pointers:
  VAR_ServerAuthentication: 0x81ab9ac
  VAR_NetVersion: 0xfd9

Offsets:
  OFFSET_RemoteSystemManager: 0x334
  OFFSET_RemoteSystemSize: 0xc69
  OFFSET_RemoteSystem__ConnectMode: 0xc62
  OFFSET_RemoteSystem__Unknown: 0xc67
=========================== FCNPC Exception Handler ============================
```

Just compiled and `https://github.com/ziggi/FCNPC/blob/master/sampsvr_files/filterscripts/traindrivers_pilots.pwn` made this to run in server

**Comment by ziggi** [comment]
Just tested it on Ubuntu 14.04 and everything works fine. Did you test it on empty gamemode? Any other plugins are used?

![image](https://github.com/ziggi/FCNPC/assets/1020099/11cad0eb-f7e1-4244-83a5-c544fd81954a)

**Comment by AtlanticCityRoleplay** [comment]
> Just tested it on Ubuntu 14.04 and everything works fine. Did you test it on empty gamemode? Any other plugins are used?
> 
> ![image](https://user-images.githubusercontent.com/1020099/262959820-11cad0eb-f7e1-4244-83a5-c544fd81954a.png)

i am still Getting Crash, I tested it on empty game mode too, 

Server: Ubuntu Server 20.04

Pawn Compiler Version: 3.10.10

Plugins Used: 

- sscanf.so
- crashdetect.so
- streamer.so 
- timerfix.so 
- pawncmd.so 
- SKY.so 
- bcrypt-samp.so 
- mysql_static.so 
- discord-connector.so 
- sampvoice_lemehost.so 
- FCNPC.so 
- nativechecker.so

I have tried FCNPC for the First Time

```
#define FILTERSCRIPT
#include <a_samp>
#include <FCNPC>

new myFirstNPC = INVALID_PLAYER_ID;

#if defined FILTERSCRIPT
public OnFilterScriptInit()
{
	myFirstNPC = FCNPC_Create("FirstNPC");
	FCNPC_Spawn(myFirstNPC, 0, 0.0, 0.0, 3.1);
	return 1;
}

public OnFilterScriptExit()
{
	FCNPC_Destroy(myFirstNPC);
	myFirstNPC = INVALID_PLAYER_ID;
	return 1;
}
#endif
```
This Example Code is too Crashing... Very Sad 😔

**Comment by ziggi** [comment]
Try to disable some plugins. I'm not sure that FCNPC are compatible with the SKY.

**Comment by AtlanticCityRoleplay** [comment]
> Try to disable some plugins. I'm not sure that FCNPC are compatible with the SKY.

Still the Same

**Comment by AtlanticCityRoleplay** [comment]
@ziggi 

Plugin is working Good in Windows Version... It seems Linux have Compatibility Issue

**Comment by ziggi** [comment]
As I said it works fine on my side. Have you tried using an empty gamemode without any plugins?

**Comment by AtlanticCityRoleplay** [comment]
> As I said it works fine on my side. Have you tried using an empty gamemode without any plugins?

Without any plugin my game mode will not run....

**Comment by AtlanticCityRoleplay** [comment]
@ziggi 

Still the Same

No Plugin Except FCNPC and No any otherFilterscript other than Example which you have given

**SERVER LOG FOR REFERENCE**

```
----------
Loaded log file: "server_log.txt".
----------

SA-MP Dedicated Server
----------------------
v0.3.7-R3, (C)2005-2019 SA-MP Team

[17:23:53] 
[17:23:53] Server Plugins
[17:23:53] --------------
[17:23:53]  Loading plugin: FCNPC.so
[17:23:53] 
[17:23:53] -------------------------------------------------
[17:23:53]      FCNPC - Fully Controllable NPC v2.0.5
[17:23:53]             GNU/Linux SA-MP 0.3.7 R2
[17:23:53]            Jul 22 2023 at 13:01:08
[17:23:53] 
[17:23:53]   Author:       OrMisicL (2013 - 2015)
[17:23:53]   Continued by: ziggi    (2016 - present)
[17:23:53] 
[17:23:53]   See full credits in the README.md file
[17:23:53] -------------------------------------------------
[17:23:53] 
[17:23:53] Loading...
[17:23:53] 
[17:23:53] -------------------------------------------------
[17:23:53]    ColAndreasv1.4.0
[17:23:53] 
[17:23:53]    Created By:
[17:23:53]      [uL]Chris42O
[17:23:53]      [uL]Slice
[17:23:53]      [uL]Pottus
[17:23:53] -------------------------------------------------
[17:23:53] 
[17:23:53] Loading...
[17:23:53] ColAndreas v1.4.0 Loaded.
[17:23:53]   Loaded.
[17:23:53]  Loaded 1 plugins.

[17:23:53] 
[17:23:53] Ban list
[17:23:53] --------
[17:23:53]  Loaded: samp.ban
[17:23:53] 
[17:23:53] 
[17:23:53] Filterscripts
[17:23:53] ---------------
[17:23:53]   Loading filterscript 'traindrivers_pilots.amx'...
```

**FCNPC CRASH REPORT**

```
=========================== FCNPC Exception Handler ============================
Signal 11 (Segmentation fault), address is 5dc30815 from 80b0036

Backtrace:
  #0 in plugins/FCNPC.so(_ZN17CExceptionHandler24ExceptionHandlerCallbackEiP9siginfo_tPv+0x126) [0xf7a3a6e6]
  #1 in [0xf7fa6b70]
  #2 in ./samp03svr() [0x80b0036]

Information:
  System: GNU/Linux
  SA-MP: 0.3.7 R2
  FCNPC: 2.0.5

Functions:
  FUNC_CPlayerPool__DeletePlayer: 0x80d0a90
  FUNC_CPlayer__Kill: 0x80cb220
  FUNC_CPlayer__EnterVehicle: 0x80cc1c0
  FUNC_CPlayer__ExitVehicle: 0x80cc340
  FUNC_CPlayer__SpawnForWorld: 0x80ccfc0
  FUNC_GetVehicleModelInfo: 0x80d5e90
  FUNC_CConsole__GetIntVariable: 0x80a0070
  FUNC_ClientJoin_RPC: 0x80b0030

Pointers:
  VAR_ServerAuthentication: 0x81ab9ac
  VAR_NetVersion: 0xfd9

Offsets:
  OFFSET_RemoteSystemManager: 0x334
  OFFSET_RemoteSystemSize: 0xc69
  OFFSET_RemoteSystem__ConnectMode: 0xc62
  OFFSET_RemoteSystem__Unknown: 0xc67
=========================== FCNPC Exception Handler ============================
```

**Comment by ziggi** [comment]
Try with 0.3.7 R3 server

**Comment by AtlanticCityRoleplay** [comment]
> Try with 0.3.7 R2 SA-MP server

Bro from the beginning i was using 0.3.7 R2 server

**Comment by AtlanticCityRoleplay** [comment]
> Try with 0.3.7 R2 SA-MP server

Ok Let me Try

**Comment by ziggi** [comment]
Can you try this one with the R3 version?
[FCNPC.zip](https://github.com/ziggi/FCNPC/files/12431783/FCNPC.zip)

**Comment by AtlanticCityRoleplay** [comment]
> Can you try this one with the R3 version?
> [FCNPC.zip](https://github.com/ziggi/FCNPC/files/12431783/FCNPC.zip)
> 

Sure let me try

**Comment by ziggi** [comment]
Should be fixed in [v2.0.6](https://github.com/ziggi/FCNPC/releases/tag/v2.0.6)
